packages/app/src/cli/utilities/app/http-reverse-proxy.ts

```diff
@@ -91,13 +100,31 @@ ${outputToken.json(JSON.stringify(rules))}
   abortController.signal.addEventListener('abort', () => {
     server.close()
   })
-  await Promise.all([
-    renderConcurrent({
-      processes: [...processes, ...additionalProcesses],
-      abortController,
-    }),
-    server.listen(availablePort),
-  ])
+
+  let renderConcurrentOptions: RenderConcurrentOptions = {
+    processes: [...processes, ...additionalProcesses],
+    abortController,
+  }
+
+  if (previewUrl) {
+    renderConcurrentOptions = {
+      ...renderConcurrentOptions,
+      onInput: (input: string) => {
+        if (input === 'p' && previewUrl) {
+          // eslint-disable-next-line @typescript-eslint/no-floating-promises
+          openURL(previewUrl)
+        } else if (input === 'q') {
+          process.exit(0)
```

Comment:
Have you tested that no sub-processes are kept alive after exiting this way?

Reply:
I haven't tested, I'll check!

Reply:
There are a few subprocesses left hanging, but the behavior is not different from pressing ctrl+c. We might need to do some other checks here.

Reply:
I've investigated, process.exit wasn't killing all the child processes so I've added the `tree-kill` package to take care of that. I've tested and it does kill everything as expected.

Reply:
🔥